File: ebegu-shared/src/main/java/ch/dvbern/ebegu/util/mandant/AbstractMandantDefaultVisitor.java
package ch.dvbern.ebegu.util.mandant;

public abstract class AbstractMandantDefaultVisitor<T> implements
	MandantVisitor<T> {

	protected abstract T visitDefault();

	@Override
	public T visitBern() {
		return visitDefault();
	}

	@Override
	public T visitLuzern() {
		return visitDefault();
	}

	@Override
	public T visitSolothurn() {
		return visitDefault();
	}

	@Override
	public T visitAppenzellAusserrhoden() {
		return visitDefault();
	}

	@Override
	public T visitSchwyz() {
		return visitDefault();
	}

	@Override
	public T visitZug() {
		return visitDefault();
	}

	@Override
	public T visitDvb() {
		return visitDefault();
	}
}

File: ebegu-web/libs/shared/model/shared-model-mandant/src/lib/AbstractMandantDefaultVisitor.ts
import {MandantVisitor} from './MandantVisitor';

export abstract class AbstractMandantDefaultVisitor<T>
    implements MandantVisitor<T>
{
    protected abstract visitDefault(): T;

    public visitAppenzellAusserrhoden(): T {
        return this.visitDefault();
    }

    public visitBern(): T {
        return this.visitDefault();
    }

    public visitLuzern(): T {
        return this.visitDefault();
    }

    public visitSchwyz(): T {
        return this.visitDefault();
    }

    public visitSolothurn(): T {
        return this.visitDefault();
    }

    public visitZug(): T {
        return this.visitDefault();
    }

    public visitDvb(): T {
        return this.visitDefault();
    }
}
